lightning/src/ln/channelmanager.rs

```diff
@@ -4352,8 +4428,13 @@ where
 		let mut random_bytes = [0u8; 16];
 		random_bytes.copy_from_slice(&self.entropy_source.get_secure_random_bytes()[..16]);
 		let user_channel_id = u128::from_be_bytes(random_bytes);
-
 		let outbound_scid_alias = self.create_and_insert_outbound_scid_alias();
+
+		// Get the number of peers with channels, but without funded ones. We don't care too much
+		// about peers that never open a channel, so we filter by peers that have at least one
+		// channel, and then limit the number of those with unfunded channels.
+		let channeled_peers_without_funding = self.peers_without_funded_channels(|node| !node.channel_by_id.is_empty());
```

Comment:
Note, when we introduce all the abstractions for collaborative transactions, we might have pending transactions with contributed inputs (i.e funded) but not broadcast yet as we have not received `tx_signatures` message. We might have to just to rename all the "funding" wording introduce by this PR, though yeah let's bikesheds when it does happen.